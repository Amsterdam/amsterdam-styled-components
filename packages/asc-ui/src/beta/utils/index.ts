import deprecatedWarning from './deprecatedWarning'

export {
  themeColor,
  srOnlyStyle,
  breakpoint,
  svgFill,
  fluidTypoStyle,
  typographyStyle,
  showAboveBackDrop,
  getValueFromTheme,
  themeSpacing,
} from './themeUtils'
export { valueFromObject } from './valueFromObject'
export { fromProps } from './fromProps'

export { deprecatedWarning }
