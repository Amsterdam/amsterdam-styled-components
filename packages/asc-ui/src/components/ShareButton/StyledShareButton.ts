import { AscCore } from '@datapunt/asc-core'
import styled from '../../styled-components'
import { getThemeColor } from '../../core/AscCoreUtils'

const StyledShareButton = styled(AscCore.Button.IconButton)<
  AscCore.Button.Props
>`
  padding: 0px;
  position: relative;
  justify-content: center;
  background: ${({ theme }) => getThemeColor(theme, 'tint', 'level5')}};

  &:focus,
  &:hover {
    background: ${({ hoverColor, theme }) =>
      hoverColor || getThemeColor(theme, 'secondary')};
  }
`

export default StyledShareButton
