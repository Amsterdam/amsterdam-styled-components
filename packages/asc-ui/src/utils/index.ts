export { default as fromTheme, valueFromObject } from './fromTheme'
export {
  color,
  getColorCode,
  getTypographyFromTheme,
  focusStyleOutline,
  srOnlyStyle,
  breakpoint,
  svgFill,
  showHide,
  ShowHideTypes,
  customCss,
  CustomCssPropsType,
  CustomCssPropsInterface,
} from './themeUtils'
