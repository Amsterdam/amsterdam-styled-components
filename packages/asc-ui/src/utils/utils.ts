import { AbstractNode, IconDefinition } from '@datapunt/asc-icons/lib/types'
import * as React from 'react'

export function log(message: string) {
  if (!(process && process.env && process.env.NODE_ENV === 'production')) {
    console.error(`[icons-react]: ${message}.`)
  }
}

export function isIconDefinition(target: any): target is IconDefinition {
  return (
    typeof target === 'object' &&
    typeof target.name === 'string' &&
    typeof target.icon === 'object'
  )
}

export function normalizeAttrs(attrs: Attrs = {}): Attrs {
  return Object.keys(attrs).reduce((acc: Attrs, key) => {
    const val = attrs[key]
    switch (key) {
      case 'class':
        acc.className = val
        delete acc.class
        break
      default:
        acc[key] = val
    }
    return acc
  }, {})
}

export interface Attrs {
  [key: string]: string
}

export class MiniMap<V> {
  get size() {
    return Object.keys(this.collection).length
  }

  private collection: { [key: string]: V } = {}

  clear(): void {
    this.collection = {}
  }

  delete(key: string): boolean {
    return delete this.collection[key]
  }

  get(key: string): V | undefined {
    return this.collection[key]
  }

  has(key: string): boolean {
    return Boolean(this.collection[key])
  }

  set(key: string, value: V): this {
    this.collection[key] = value
    return this
  }
}

export function generate(
  node: AbstractNode,
  key: string,
  rootProps?: { [key: string]: any } | false,
): any {
  if (!rootProps) {
    return React.createElement(
      node.tag,
      { key, ...normalizeAttrs(node.attrs) },
      (node.children || []).map((child, index) =>
        generate(child, `${key}-${node.tag}-${index}`),
      ),
    )
  }
  return React.createElement(
    node.tag,
    {
      key,
      ...normalizeAttrs(node.attrs),
      ...rootProps,
    },
    (node.children || []).map((child, index) =>
      generate(child, `${key}-${node.tag}-${index}`),
    ),
  )
}
