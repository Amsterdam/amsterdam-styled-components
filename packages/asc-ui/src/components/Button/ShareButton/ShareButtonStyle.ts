import styled from '@datapunt/asc-core'
import IconStyle from '../../Icon/IconStyle'
import { color, svgFill } from '../../../utils'
import ButtonStyle, { Props as ButtonProps } from '../ButtonStyle'

export type Props = {
  hoverColor?: string
} & ButtonProps

const ShareButtonStyle = styled(ButtonStyle)<Props>`
  padding: 0;
  position: relative;
  justify-content: center;
  background: ${color('tint', 'level5')};

  &:focus,
  &:hover {
    background: ${({ hoverColor }) => hoverColor || color('secondary')};
  }

  ${IconStyle} {
    ${svgFill('tint', 'level1')};
  }
`

export default ShareButtonStyle
