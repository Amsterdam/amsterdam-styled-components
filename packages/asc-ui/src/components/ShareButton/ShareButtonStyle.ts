import styled from '@datapunt/asc-core'
import IconStyle from '../Icon/IconStyle'
import IconButtonStyle, {
  Props as IconButtonStyleProps,
} from '../IconButton/IconButtonStyle'
import { color, svgFill } from '../../utils'

export type Props = IconButtonStyleProps

const ShareButtonStyle = styled(IconButtonStyle)<Props>`
  padding: 0;
  position: relative;
  justify-content: center;
  background: ${color('tint', 'level5')};

  &:focus,
  &:hover {
    background: ${({ hoverColor }) => hoverColor || color('secondary')};
  }

  ${IconStyle} {
    ${({ theme }) => svgFill('tint', 'level1')({ theme })};
  }
`

export default ShareButtonStyle
