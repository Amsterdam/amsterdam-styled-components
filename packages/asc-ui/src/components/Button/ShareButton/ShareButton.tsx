import React from 'react'
import Icon from '../../Icon'
import ShareButtonStyle, {
  Props as ShareButtonStyleProps,
} from './ShareButtonStyle'

type Props = {
  iconSize?: number
} & ShareButtonStyleProps &
  React.HTMLAttributes<HTMLButtonElement>

const ShareButton: React.FC<Props> = ({
  children,
  onClick,
  iconSize,
  ...otherProps
}) => (
  <ShareButtonStyle
    data-testid="share-button"
    size={30}
    onClick={onClick}
    variant="blank"
    {...otherProps}
  >
    <Icon size={iconSize}>{children}</Icon>
  </ShareButtonStyle>
)

export default ShareButton
