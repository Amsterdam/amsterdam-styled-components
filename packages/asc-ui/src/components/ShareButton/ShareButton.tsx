import React from 'react'
import { Icon } from '../..'
import ShareButtonStyle, {
  Props as ShareButtonStyleProps,
} from './ShareButtonStyle'

type Props = {
  onClick?: React.MouseEventHandler<HTMLButtonElement>
  iconSize?: number
} & ShareButtonStyleProps

const ShareButton: React.FC<Props> = ({
  children,
  onClick,
  iconSize,
  ...otherProps
}) => (
  <ShareButtonStyle onClick={onClick} square {...otherProps}>
    <Icon size={iconSize}>{children}</Icon>
  </ShareButtonStyle>
)

export default ShareButton
