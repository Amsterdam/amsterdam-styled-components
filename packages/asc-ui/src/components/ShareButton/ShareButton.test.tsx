import * as React from 'react'
import ShareButton from './ShareButton'
import { renderWithTheme } from '../../utils/withTheme'
import 'jest-styled-components'

describe('ShareButton', () => {
  it('should render the share button', () => {
    const component = renderWithTheme(<ShareButton>Hello</ShareButton>)

    expect(component).toMatchSnapshot()
  })
})
