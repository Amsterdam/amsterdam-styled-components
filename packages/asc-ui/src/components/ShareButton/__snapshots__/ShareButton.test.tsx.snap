// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShareButton should render the share button 1`] = `
.c1 {
  display: block;
  width: 20px;
  height: 20px;
  padding: undefinedpx;
  box-sizing: content-box;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

.c1 > svg {
  width: inherit;
  height: inherit;
}

.c1 > svg rect,
.c1 > svg polygon,
.c1 > svg path {
  fill: #fff;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  border: none;
  cursor: pointer;
  font-size: 16px;
  font-weight: normal;
  padding: 0 10px 0 10px;
  -webkit-transition: color 0.1s ease-in-out,background-color 0.1s ease-in-out;
  transition: color 0.1s ease-in-out,background-color 0.1s ease-in-out;
  background: #ffffff;
  padding: 5px;
  width: 30px;
  height: 30px;
  padding: 0px;
  position: relative;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: #767676;
}

.c0:focus {
  outline-color: #FEC813;
  outline-style: solid;
  outline-width: medium;
}

.c0:hover {
  background: #E6E6E6;
}

.c0:after {
  content: '';
  display: block;
  min-height: inherit;
  font-size: 0;
}

.c0:focus,
.c0:hover {
  background: #ec0000;
}

<button
  class="c0"
>
  <span
    class="c1"
    color="secondary"
  >
    Hello
  </span>
</button>
`;
